llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -59,7 +59,7 @@ Value *InstCombinerImpl::insertRangeTest(Value *V, const APInt &Lo,
 
   // V >= Min && V <  Hi --> V <  Hi
   // V <  Min || V >= Hi --> V >= Hi
-  ICmpInst::Predicate Pred = Inside ? ICmpInst::ICMP_ULT : ICmpInst::ICMP_UGE;
+  CmpPredicate Pred = Inside ? ICmpInst::ICMP_ULT : ICmpInst::ICMP_UGE;
```

Comment:
I'd probably leave this one alone, as this is just holding a plain predicate...

Reply:
It is required for typing of the ternary operator a few lines below:

```
    Pred = isSigned ? ICmpInst::getSignedPredicate(Pred) : Pred;
```